src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -949,3 +949,25 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
         elif hostname in hostnames_in_webapp:
             matched.add(hostname)
     return matched
+
+
+def create_function(resource_group_name, name, storage_account, plan=None):
+    client = web_client_factory()
+    if is_valid_resource_id(plan):
+        plan = parse_resource_id(plan)['name']
+    location = _get_location_from_app_service_plan(client, resource_group_name, plan)
+    webapp_def = Site(server_farm_id=plan, location=location)
+    webapp_def.kind = 'functionapp'
+    sa_resource_group = resource_group_name
+    if is_valid_resource_id(storage_account):
+        sa_resource_group = parse_resource_id(storage_account)['resource_group']
```

Comment:
that wont work storage account can be passed as a resource id if account is in different resource group 
eg . -s /subscriptions/xxxx/resourceGroups/yyyy/providers/Microsoft.Storage/storageAccounts/zzz
and also like this if the storage account is in the same resource group
-s zzz